Remove documentation of the abandoned transformation rule for disjointness

gkostkowski opened the pull request:
Documentation of the rule has been removed along with a related convention. Other related part of the documentation has been updated.

Dragos0000 reviewed (approved):
change  folder name from old-docs to archived-docs

gkostkowski committed:
Remove documentation of the abandoned transformation rule and a related convention rule, update documentation

The rule R.18 was implementing incorrect behaviour due to wrong assumption about
UML semantics. Subclasses are not disjoint if not explicitly marked as such with
the `isDisjoint` generalization set property (that are not currently supported by
model2owl).
